### PR TITLE
scaffold for rds db creation

### PR DESCRIPTION
**Pull request by DaedalusG**
This introduces the option to create three RDS instances while provisioning a cluster for AWS EKS. Intended for use in testing Sourcegraph with external RDS databases.

### CODE REVIEW

File: aws/terraform.tfvars [hunk 1]
```diff
@@ -1,11 +1,29 @@
...
 node_group_one_instance_types = ["m6a.4xlarge"]
-node_group_one_min_size = 2
-node_group_one_max_size = 6
-node_group_one_desired_size = 4
+node_group_one_min_size       = 2
+node_group_one_max_size       = 6
+node_group_one_desired_size   = 4
+
+### frontend db connection strings ###
+frontend_db_name              = "frontend"
```
Comment (loujar) [inline aws/terraform.tfvars]: could we change this to `sg` for consistency with our helm chart?

Comment (loujar) [reply aws/terraform.tfvars]: for reference, these are the defaults from helm:
```
- PGHOST=pgsql
- PGPORT=5432
- PGUSER=sg
- PGPASSWORD=sg
- PGDATABASE=sg
- PGSSLMODE=disable

- CODEINTEL_PGHOST=codeintel-db
- CODEINTEL_PGPORT=5432
- CODEINTEL_PGUSER=sg
- CODEINTEL_PGPASSWORD=sg
- CODEINTEL_PGDATABASE=sg
- CODEINTEL_PGSSLMODE=disable

- CODEINSIGHTS_PGHOST=codeinsights-db
- CODEINSIGHTS_PGPORT=5432
- CODEINSIGHTS_PGUSER=postgres
- CODEINSIGHTS_PGPASSWORD=password
- CODEINSIGHTS_PGDATABASE=postgres
- CODEINSIGHTS_PGSSLMODE=disable
```

File: aws/variables.tf [hunk 2]
```diff
@@ -31,3 +31,107 @@ variable "node_group_one_desired_size" {
   type        = number
   default     = 4
 }
+
+### RDS Switch ###
+variable "create_rds" {
```
Comment (loujar) [inline aws/variables.tf]: missing description

File: aws/variables.tf [hunk 3]
```diff
@@ -31,3 +31,107 @@ variable "node_group_one_desired_size" {
...
+
+### codeinsights DB ###
+variable "codeinsights_db_name" {
+  description = "Name for the codeintel database"
+  type        = string
+  default     = "codeintel"
+}
+
+variable "codeinsights_db_username" {
+  description = "Name for the codeintel database"
```
Comment (loujar) [inline aws/variables.tf:130]: typo, should say "codeinsights database"

File: aws/variables.tf [hunk 4]
```diff
@@ -31,3 +31,107 @@ variable "node_group_one_desired_size" {
...
+variable "codeintel_db_allocated_storage" {
+  description = "DB allocated storage"
+  type        = number
+  default     = 100
+}
+
+
+### codeinsights DB ###
+variable "codeinsights_db_name" {
+  description = "Name for the codeintel database"
```
Comment (loujar) [inline aws/variables.tf:124]: typo, should say "codeinsights database"

File: aws/variables.tf [hunk 5]
```diff
@@ -31,3 +31,107 @@ variable "node_group_one_desired_size" {
...
+variable "create_rds" {
+  type    = bool
+  default = false
+}
+
+### RDS DB connection strings
+
+### Frontend DB ###
+variable "frontend_db_name" {
+  description = "Name for the codeintel database"
```
Comment (loujar) [inline aws/variables.tf:48]: typo, should say "frontend database"

File: aws/variables.tf [hunk 6]
```diff
@@ -31,3 +31,107 @@ variable "node_group_one_desired_size" {
...
+  type        = string
+  default     = "frontend"
+}
+
+variable "frontend_db_username" {
+  description = "Name for the codeintel database"
```
Comment (loujar) [inline aws/variables.tf:54]: typo, should say "frontend database"

**Comment by loujar** [comment]
just a thought, we might want to adjust the description for this repo since we're expanding the scope of its functionality: 
> Terraform configurations for provisioning bare bones Kubernetes clusters on GCP, AWS and Azure platforms.

**Comment by DaedalusG** [comment]
@jdpleiness  I'd done the same stuff locally and just hadn't committed yet lol

Also I removed the default passwords in tfvars so that it forces a password prompt for the security conscientious

**Comment by jdpleiness** [comment]
> I'd done the same stuff locally and just hadn't committed yet lol

My bad, I was in there testing the deployments and just patched it real quick when I came across the issue 😬 

> Also I removed the default passwords in tfvars so that it forces a password prompt for the security conscientious

I don't think this is an issue as we have these as defaults across our repos, with the expectation that customers will change them in a production deployment.

We could add a notice in the `README.md` or the `.tfvars` file about this as well.

**Comment by DaedalusG** [comment]
@jdpleiness I think I'm going to set up an option to just provision one rds database to host all our dbs as well, but this is out of scope atm. wdyt merge for now and come back if we prioritize this sort of env setup tooling?

File: aws/variables.tf (hunk 5 above)
Comment (jasonhawkharris) [reply aws/variables.tf:48]: lol what a n00b amirite? @DaedalusG

**Review by jdpleiness — APPROVED** [review]

**Pull request merged**